spec/models/configuration_script_base_spec.rb

```diff
@@ -0,0 +1,13 @@
+describe ConfigurationScriptBase do
+  it "has many authentications and vice versa" do
```

Comment:
> But trusting other developers' tests to cover your code seems shaky to me.

Who said you were going to trust another developer's tests? ^_^ You're just modeling things now but I would expect whoever is continuing work on this feature would write specs to cover the feature which would utilize said associations and prove the point I'm making. 

> then we should remove a bunch existing specs e.g. here...

You aren't wrong. There probably are such things that might be removed. Note though that a cursory glance at the example you give yields that the assigning of `configuration_scripts` to set up the correct scenario for those specs is not trivial (see factories and all the metaprogramming around those specs to see what I mean). Sometimes when setup of a proper scenario to test something is complex, tests are written that ensure the scenario is set up accurately before other tests are run. It could be that specs were failing with arcane error messages because the scenario was set up incorrectly, and thus he added seemingly unimportant tests to ensure that if the configuration scripts weren't set up correctly for all the rest of the tests, we'd see something very clear as to why that's the case.

The assignments in this PR are trivial and don't test anything beyond Rails associations, is my point.

We've spent a ton of time discussing this so I'll be dropping it after this point. The main thing I'd like to stress is that the only reason I would block this from being merged is that it needlessly creates database writes, slowing down our test suite, when the thing you're trying to test can be done in memory (again, if you care to test it at all, which is something I personally would not do). 

Cool?